### ISSUE TITLE
Results from model not matching table

### ISSUE THREAD

**Issue by jntorres** [antonior92/automatic-ecg-diagnosis#4]
Hello All, 

I really enjoyed your paper and wanted to see how I can reproduce the results from Table 2. I downloaded the model and test dataset. When I predict the classes using model.predict and use a threshold of 0.5 I get the following breakdown of precision, recall and f1-scores. What were the thresholds used to get the results in Table 2? Am I missing any steps taken? 

Thank you!

```

classification report

              precision    recall  f1-score   support

           0       1.00      0.25      0.40        28
           1       0.94      0.85      0.89        34
           2       1.00      0.83      0.91        30
           3       0.86      0.75      0.80        16
           4       1.00      0.69      0.82        13
           5       0.93      0.76      0.84        37

   micro avg       0.95      0.70      0.80       158
   macro avg       0.95      0.69      0.78       158
weighted avg       0.96      0.70      0.78       158
 samples avg       0.13      0.12      0.13       158

```

**Comment by antonior92** [comment]
Hi @jntorres. The training dataset we used to train the model is highly unbalanced. See this table
[here](https://www.nature.com/articles/s41467-020-15432-4/tables/1).

So we compensate for this unbalanced data using threshold different than 0.5. We find the threshold value that gives the best F1-score: see [here](https://github.com/antonior92/automatic-ecg-diagnosis/blob/16dbb166a83b88e2e7fba314c8082011462153a8/generate_figures_and_tables.py#L28)

This paper here has some discussion about how to deal with class imbalance: https://arxiv.org/pdf/1710.05381.pdf. We are using some variant of the version he calls ``thresholding'' in section 2.2. Where, instead of using the proportions of the classes to set the threshold, we set it based on the data and the precision-recall curve.

**Comment by jntorres** [comment]
Thanks for the clarification. I am assuming the thresholds selected for each class determined by the validation set?

**Comment by antonior92** [comment]
Yes that is it. 

The code [here](https://github.com/antonior92/automatic-ecg-diagnosis/blob/16dbb166a83b88e2e7fba314c8082011462153a8/generate_figures_and_tables.py#L28) might be slightly misleading because it shows it is being done in the test set and not on the validation set. I will fix that.

Anyway, the result is basically the same in the two cases. Mostly, as you can see in the precision-recall curve:
![precision-recall](https://www.nature.com/articles/s41467-020-15432-4/figures/2)
there are some relative large plateaus.
